docs/vocabulary.md

```diff
@@ -388,6 +389,230 @@ pipeline in corresponding documentation.
+* Semantics: Provision build resources, set up the build workspace.
+* Aliases: Initialize, Start, Prepare, Workspace
+* Inputs: Build Request Parameters, Build Container Image Name and Version
+* Outputs: Secrets, Build Environment, Build Workspace
+* Other Results and Side Effects: N/A
+
+#### Source
+* Semantics: Download, retrieve or copy software, images, and documentation into the build workspace. Fetch configuration data.
+* Aliases: Clone, Fetch
+* Inputs: Source Code Reference (Repository, Branch, Commit)
+* Outputs: Software Source, Binary Source
+* Other Results and Side Effects: N/A
+
+#### Tag
+* Semantics: Annotate source code, images, etc. with information, such as the version number and a description.
+* Aliases: Annotate, Version
+* Inputs: Source Code Reference (Repository, Branch, Commit), Software Source, Binary Source, Generated Software, Generated Binaries
+* Outputs: Tag/annotation metadata
+* Other Results and Side Effects: Tags/annotations have been added to the software.
+
+#### Secret Detection
+* Semantics: Detect secrets in the source code, other software, or documentation. Examples include passwords, SSH keys, API keys, and so on.
+* Aliases: N/A
+* Inputs: Source Code Reference (Repository, Branch, Commit), Software Source, Binary Source
+* Outputs: Secret Detection Report
+* Other Results and Side Effects: Revoked Secrets
+
+#### Build
+* Semantics: Assemble and/or compile software and documentation into an executable and usable format.
+* Aliases: Compile, Install, Assemble, Generate
+* Inputs: Software Source, Binary Source
+* Outputs: Generated Software, Generated Binaries
+* Other Results and Side Effects: N/A
+
+#### Discovery
+* Semantics: Perform deep discovery to identify all dependencies (including transitive dependencies) in the software and documentation as packaged.
+* Aliases: Dependencies
+* Inputs: Software Source, Binary Source, Generated Software, Generated Binaries
+* Outputs: Dependency List/Graph
+* Other Results and Side Effects: N/A
+
+#### Remediate
+* Semantics: Find and automatically fix known vulnerabilities for application package dependencies, container base images and os packages.
+* Aliases: Fix, Update
+* Inputs: Software Source, Binary Source, Generated Software, Generated Binaries, Source Code Reference (Repository, Branch, Commit), Dependency List/Graph
+* Outputs: Remediated Software or Documentation, Container Images, Binaries, Source Code Reference (Repository, Branch, Commit), Dependency List/Graph
+* Other Results and Side Effects: May also update the Source Code Repository with new dependencies, create a pull request with the updates, or open an issue requesting the updates.
+
+#### Test
+* Semantics: Run a test suite. Examples includes unit tests, integration tests, acceptance tests, performance tests, canary tests, A/B tests, smoke tests.
+* Aliases: Validate
+* Inputs: Any
+* Outputs: Test Results, Test Reports, Test Coverage Reports
+* Other Results and Side Effects: N/A
```

Comment:
The same way that we can generate a test coverage report that we post-analyze to see if we should fail it or not, there are probably quite a lot of other things that could be output by test that we want to post-analyze.

A common example would be that the test produces a log file from an application being tested, and we want to post-analyze that log file to extract some flow of internal events that we then use as input for some other test.

Most likely we could say that we are then still in the test stage, which is fine, but then I wonder what is special about Coverage that makes it appropriate for its own stage.

Reply:
I think you're right, and we can merge the two.  Coverage can be a type of a test - namely, testing whether the coverage was enough.